Gems/ROS2/Code/Source/VehicleDynamics/DriveModels/AckermannDriveModel.cpp

```diff
@@ -0,0 +1,168 @@
+            return;
+        }
+
+        auto innerSteering = atan(
+            (m_vehicleConfiguration.m_wheelbase * tan(steering)) /
+            (m_vehicleConfiguration.m_wheelbase - 0.5 * m_vehicleConfiguration.m_track * tan(steering)));
+        auto outerSteering = atan(
+            (m_vehicleConfiguration.m_wheelbase * tan(steering)) /
+            (m_vehicleConfiguration.m_wheelbase + 0.5 * m_vehicleConfiguration.m_track * tan(steering)));
+
+        ApplyWheelSteering(m_steeringData.front(), innerSteering, deltaTimeNs);
+        ApplyWheelSteering(m_steeringData.back(), outerSteering, deltaTimeNs);
+    }
+
+    void AckermannDriveModel::ApplySpeed(float speed, uint64_t deltaTimeNs)
+    {
+        if (m_disabled)
+        {
+            return;
+        }
+        if (m_driveWheelsData.empty())
+        {
+            AZ_Warning("ApplySpeed", false, "Cannot apply speed since no diving wheels are defined in the model");
+            return;
+        }
+
+        const double deltaTimeSec = double(deltaTimeNs) / 1e9;
+        for (const auto& wheelData : m_driveWheelsData)
+        {
+            auto wheelEntity = wheelData.m_wheelEntity;
+            AZ::Transform wheelTransform;
+            AZ::TransformBus::EventResult(wheelTransform, wheelEntity, &AZ::TransformBus::Events::GetWorldTM);
+
+            AZ::Transform inverseWheelTransform = wheelTransform.GetInverse();
+            AZ::Vector3 currentAngularVelocity;
+            Physics::RigidBodyRequestBus::EventResult(currentAngularVelocity, wheelEntity, &Physics::RigidBodyRequests::GetAngularVelocity);
+            currentAngularVelocity = inverseWheelTransform.TransformVector(currentAngularVelocity);
+            auto currentAngularSpeedX = currentAngularVelocity.Dot(wheelData.m_driveAxis);
+            float wheelRadius = wheelData.m_wheelRadius;
+            if (AZ::IsClose(wheelRadius, 0.0f))
+            {
+                const float defaultFloatRadius = 0.35f;
+                AZ_Warning("ApplySpeed", false, "Wheel radius is zero (or too close to zero), resetting to default %f", defaultFloatRadius);
+                wheelRadius = defaultFloatRadius;
+            }
+
+            auto desiredAngularSpeedX = speed / wheelRadius;
+            double pidCommand = m_speedPid.ComputeCommand(desiredAngularSpeedX - currentAngularSpeedX, deltaTimeNs);
+            if (AZ::IsClose(pidCommand, 0.0)) // TODO - use the third argument with some reasonable value which means "close enough"
+            {
+                continue;
+            }
+
+            auto impulse = pidCommand * deltaTimeSec;
+
+            auto transformedTorqueVector = wheelTransform.TransformVector(wheelData.m_driveAxis * impulse);
+            Physics::RigidBodyRequestBus::Event(wheelEntity, &Physics::RigidBodyRequests::ApplyAngularImpulse, transformedTorqueVector);
+        }
+    }
+
+    void AckermannDriveModel::SetDisabled(bool is_disabled)
+    {
+        m_disabled = is_disabled;
+    }
```

Comment:
`is_disabled` should be lowerCamelCase per the [coding standards and style guide](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#variable-names)
```suggestion
    void AckermannDriveModel::SetDisabled(bool isDisabled)
    {
        m_disabled = isDisabled;
    }
```